code/src/sixsq/nuvla/server/resources/spec/notification.cljc

```diff
@@ -0,0 +1,92 @@
+(ns sixsq.nuvla.server.resources.spec.notification
+  (:require
+    [clojure.spec.alpha :as s]
+    [sixsq.nuvla.server.resources.spec.common :as common]
+    [sixsq.nuvla.server.resources.spec.core :as core]
+    [sixsq.nuvla.server.util.spec :as su]
+    [spec-tools.core :as st]))
+
+
+(s/def ::message
+  (-> (st/spec ::core/nonblank-string)
+      (assoc :name "message"
+             :json-schema/description "notification text"
+
+             :json-schema/order 30)))
+
+
+(s/def ::type
+  (-> (st/spec ::core/kebab-identifier)
+      (assoc :name "type"
+             :json-schema/description "notification type"
+
+             :json-schema/order 31)))
+
+
+(s/def ::content-unique-id
+  (-> (st/spec ::core/nonblank-string)
+      (assoc :name "content-unique-id"
+             :json-schema/display-name "notification content unique id"
+             :json-schema/description "notification content unique id"
+
+             :json-schema/order 32)))
+
+
+(s/def ::expiry
+  (-> (st/spec ::core/timestamp)
+      (assoc :name "expiry"
+             :json-schema/description "notification expiry"
+
+             :json-schema/order 33)))
+
+
+(s/def ::hide-until
```

Comment:
I'm now wondering whether "not-before" would be a better name for this field. It is more in keeping with the fields used for managing the validity of certificates. It also doesn't imply that something will be done other than updating the field; "hide-until" is implying slightly more than it actually does.  Not too fussed either way, but better to change it now before it gets frozen in place, if we do change it.

Reply:
Agreed.